src/pages/TeamPage/TeamPage.tsx

```diff
@@ -74,9 +74,9 @@ export const TeamPage: React.FC = () => {
     // TODO: review; there's is something wrongly typed somewhere
     if (target.value) {
       dispatch(changeRole({
-        org_id: orgId.toString(),
-        user_id: userId.toString(),
-        role_id: target.value,
+        org_id: orgId,
+        user_id: `${userId.toString()}`,
+        role_id: `${ROLES[target.value].level}`,
```

Comment:
This is not correct. The role sent should be the value and not the level. 
The levels used in the FE are FE use only, to locally assign the hierarchy. It can be a match with BE ids but this is not always true.